examples/encoding/02_simple_encoding_dash_cenc_hls_fairplay.js

```diff
@@ -0,0 +1,655 @@
+        codecConfigurationH264At272p,
+        codecConfigurationAAC
+      ]) => {
+      console.log('Successfully created input, output and codec configurations.');
+      const dashManifestCreationPromise = createDashManifestWithPeriodAndAdaptationSets(output.id);
+      const hlsManifestCreationPromise = createHlsManifest(output.id);
+      Promise.all([dashManifestCreationPromise, hlsManifestCreationPromise]).then(([dashManifest, hlsManifest]) => {
+        console.log('Successfully created DASH and HLS Manifest Resources');
+        [createdDashManifest, createdDashManifestPeriod, createdDashManifestAudioAdaptationSet, createdDashManifestVideoAdaptationSet] = dashManifest;
+
+        const streamDefinitions = [
+          {
+            codecConfiguration: codecConfigurationAAC,
+            adaptationSet: createdDashManifestAudioAdaptationSet,
+            outputPathDiscriminator: 'audio/128'
+          },
+          {
+            codecConfiguration: codecConfigurationH264At816p,
+            adaptationSet: createdDashManifestVideoAdaptationSet,
+            outputPathDiscriminator: 'video/816'
+          },
+          {
+            codecConfiguration: codecConfigurationH264At544p,
+            adaptationSet: createdDashManifestVideoAdaptationSet,
+            outputPathDiscriminator: 'video/544'
+          },          {
+            codecConfiguration: codecConfigurationH264At272p,
+            adaptationSet: createdDashManifestVideoAdaptationSet,
+            outputPathDiscriminator: 'video/272'
+          }
+        ];
+
+        const promiseMap = Promise.map(streamDefinitions, (streamDefinition) => {
+          return createStreamWithMuxingsAndDRMsAndManifestResources(
+            streamDefinition.codecConfiguration, input, output, streamDefinition.outputPathDiscriminator, encoding,
+            createdDashManifest, createdDashManifestPeriod, streamDefinition.adaptationSet, hlsManifest
+          );
+        }, {concurrency: 1});
+
+        promiseMap.then((result) => {
+          console.log('Successfully created all resources. Starting Encoding Process...');
+          startEncodingAndWaitForItToBeFinished(encoding).then(() => {
+            console.log('Successfully Finished Encoding Process.. Starting manifest creation...');
+            const dashCreationPromise = startDashManifestCreationAndWaitForItToBeFinished(createdDashManifest);
+            const hlsCreationPromise = startHlsManifestCreationAndWaitForItToBeFinished(hlsManifest);
+            Promise.all([dashCreationPromise, hlsCreationPromise]).then(() => {
+              console.log('Successfully finished DASH and HLS manifest creation!');
+              resolve(result);
+            });
+          });
+        }).catch((error) => {
+          console.error('An error occured: ', error);
+          reject(error);
+        });
+      });
+    });
+  });
+};
+
+const createInput = () => {
+  return new Promise((resolve, reject) => {
```

Comment:
You can cut down the number of promises we throw around if we just do the following:

```
return bitmovin.encoding.inputs.http.create(httpInput).then((createdInput) => {
  console.log(...);
  return createdInput;
}).catch((error) => {
  console.log(...);
  return error;
});
```

Thus saving 2 lines on each promise that is only doing logging..